### PR TITLE
fix(editable-html): hide DoneButton on active htmlMode, Prevent infinite loop in componentDidUpdate by refining isEdited state check PD-2765

### PR DESCRIPTION
**Pull request by CarlaCostea**
https://illuminate.atlassian.net/browse/PD-2765

### CODE REVIEW

**Comment by andreeapescar** [comment]
@CarlaCostea would you mind adding some comments to simply explain the logic in those if's? it's hard to follow.

**Comment by CarlaCostea** [comment]
@andreeapescar  I've added clarifying comments to those if conditions to make the logic more understandable. Please let me know if they provide enough clarity or if further explanations are needed.

**Review by andreeimiron — APPROVED** [review]

**Review by arimieandreea — APPROVED** [review]
